Source/WebKit/UIProcess/Cocoa/NavigationState.mm

```diff
@@ -287,7 +287,7 @@
     if (!navigationDelegate)
         return;
 
-    [static_cast<id<WKNavigationDelegatePrivate>>(navigationDelegate) _webViewWillEndNavigationGesture:webView().get() withNavigationToBackForwardListItem:willNavigate ? wrapper(item) : nil];
+    [static_cast<id<WKNavigationDelegatePrivate>>(navigationDelegate) _webViewWillEndNavigationGesture:webView().get() withNavigationToBackForwardListItem:RetainPtr { willNavigate ? wrapper(item) : nil }.get()];
```

Comment:
Should we change the `wrapper` functions to return `RetainPtr`? When would we ever **not** want to use `protectedWrapper`? For that matter, why doesn’t this use `protectedWrapper`? Also pretty sure we can write either `willNavigate ? RetainPtr { wrapper(item) }.get() : nil` or `willNavigate ? protectedWrapper(item).get() : nil` and the value is protected until the end of the evaluation of the full expression (after the method returns). The only tricky exception to this rule that I am aware of that we have to beware is with a range-based for loop, where the end of the full expression is before the for loop starts!

Reply:
Sure, in theory, but are there any cases like this?